proxy/src/main/java/com/wavefront/agent/logsharvesting/FilebeatListener.java

```diff
@@ -0,0 +1,185 @@
+package com.wavefront.agent.logsharvesting;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import com.wavefront.agent.PointHandler;
+import com.wavefront.agent.config.LogsIngestionConfig;
+import com.wavefront.agent.config.MetricMatcher;
+import com.yammer.metrics.core.Counter;
+import com.yammer.metrics.core.Gauge;
+import com.yammer.metrics.core.Metric;
+import com.yammer.metrics.core.MetricName;
+import com.yammer.metrics.core.MetricsRegistry;
+
+import org.logstash.beats.IMessageListener;
+import org.logstash.beats.Message;
+
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import io.netty.channel.ChannelHandlerContext;
+import sunnylabs.report.ReportPoint;
+import sunnylabs.report.TimeSeries;
+
+/**
+ * Listens for messages from Filebeat and processes them, sending telemetry to a given sink.
+ *
+ * @author Mori Bellamy (mori@wavefront.com)
+ */
+public class FilebeatListener implements IMessageListener {
+  protected static final Logger logger = Logger.getLogger(FilebeatListener.class.getCanonicalName());
+  private static final FlushProcessor flushProcessor = new FlushProcessor();
+  private static ReadProcessor readProcessor = new ReadProcessor();
+  private final PointHandler pointHandler;
+  private final MetricsRegistry metricsRegistry;
+  private final LogsIngestionConfig logsIngestionConfig;
+  private final LoadingCache<MetricName, Metric> metricCache;
+  private final Counter received, evicted, unparsed, batchesProcessed, parsed;
+  private final String hostname, prefix;
+
+  /**
+   * @param pointHandler        Play parsed metrics and meta-metrics to this
+   * @param logsIngestionConfig configuration object for logs harvesting
+   * @param hostname            hostname for meta-metrics (so, for this proxy)
+   * @param prefix              all harvested metrics start with this prefix
+   * @param reportMetaMetrics   if true, report meta-metrics. parameterized so we can turn off for testing
+   */
+  public FilebeatListener(PointHandler pointHandler, LogsIngestionConfig logsIngestionConfig, String hostname,
+                          String prefix, boolean reportMetaMetrics) {
+    this.pointHandler = pointHandler;
+    this.hostname = hostname;
+    this.prefix = prefix;
+    this.logsIngestionConfig = logsIngestionConfig;
+    this.metricsRegistry = new MetricsRegistry();
+    // Meta metrics.
+    this.received = metricsRegistry.newCounter(FilebeatListener.class, "received");
```

Comment:
Is there a reason why these metrics are sent outside of the usual "agent metrics" flow? There's also a possibility that it might prevent them from being replicated to 'mon' instance to be monitored centrally.

`this.received = Metrics.newCounter(new MetricName("logsharvesting", "", "received"));`
would automatically register this metric and it will be sent along with the rest during agent check-in, so a separate scheduled task would be not necessary. 
If we do need to send these particular metrics more frequently (like every second), should we move them to a different namespace?